src/crypto/tls/common.go

```diff
@@ -673,8 +703,22 @@ type Config struct {
 	// used for debugging.
 	KeyLogWriter io.Writer
 
+	// SupportDelegatedCredential is true if the client or server is willing
+	// to negotiate the delegated credential extension.
+	// This can only be used with TLS 1.3.
+	//
+	// See https://tools.ietf.org/html/draft-ietf-tls-subcerts.
+	SupportDelegatedCredential bool
+
+	// GetDelegatedCredential returns a DelegatedCredential for use with the
+	// delegated credential extension based on the ClientHello and TLS version
+	// selected for the session. If this is nil, then the server will not offer
+	// a DelegatedCredential.
```

Comment:
```suggestion
	// GetDelegatedCredential returns a DelegatedCredential for use with the
	// delegated credential extension based on the ClientHello and TLS version
	// selected for the session. If this is nil, then the server will not offer
	// a DelegatedCredential.
	// If the returned DelegatedCredential is nil, it will also not offer a 
	// delegated credential.
```

Seems to be a good idea to document this ability to not use a DC if the client doesn't advertise the right supported algorithms.

Reply:
I added some new text.